test/integration/cleanliness_test.rb

```diff
@@ -11,14 +11,4 @@
     # if it is not avoidable see http://stackoverflow.com/questions/7163264
     File.read('Gemfile.lock').wont_include 'rails-assets-bootstrap '
   end
-
-  if ActiveRecord::Base.connection.adapter_name == "Mysql2"
-    it "uses the right row format in mysql" do
```

Comment:
you're right, if utf8mb4 is not used then the index size limits won't be exceeded. It did take us a while to figure out why migrations were failing and how to fix it though. Is it reasonable to use the monkey patch in the test environment and keep this spec, but make it conditional on an envar?